kodilla-patterns2/src/main/java/com/kodilla/patterns2/adapter/bookclasifier/libraryb/Statistics.java

```diff
@@ -17,12 +17,11 @@ public int averagePublicationYear(Map<BookSignature, Book> books) {
     @Override
     public int medianPublicationYear(Map<BookSignature, Book> books) {
         if (books.size() == 0) return 0;
-        int[] years = new int[books.size()];
-        int n = 0;
-        for (Map.Entry<BookSignature, Book> entry : books.entrySet()) {
-            years[n] = entry.getValue().getYearOfPublication();
-            n++;
-        }
+        Integer[] years = Arrays.stream(books.values()
```

Comment:
A co tu dziwnego sie dzieje?

Reply:
Robię stream mapy books i tworzę tablicę years. Streamuję mapę po Value, następnie mapuję na właściwość yearOfPublication. Otrzymuję stream Integerów. Następnie zbieram je jako tablica Integerów. I w tym momencie jeśli bym zostawił, to dostaję błąd: 

Incompatible types. Required Integer[] but stream was inferred to Stream<T>: no instance of type variable <T> exists so that stream<T> conforms to Integer[T]

Dlatego jeszcze transformuję na koniec do tablicy Integerów.